### DISCUSSION TITLE
Messaging between widgets

### DISCUSSION THREAD

**Discussion by presedo93** [Textualize/textual#186] [Q&A]
Hi!

I'm playing with textual to build an awesome dashboard for a deep learning training script! But I'm struggling a little bit with the way to communicate between Widgets. I've found the way to communicate a children with its parent. But is there any way to communicate between Widgets that are "brothers" or are even located at different leves?

What I have is an Input Text widget and I want every time I press the "enter" button that the content of text box is sent to other widget... I've tried with `post_message` but I can't receive the message in the other Widget...

Thanks!!

**Comment by willmcgugan** [comment]
You want `emit` which will send the message to the parent. You could post a message directly to a sibling with `sibling.post_message`, but it's a more scalable design to do that via the parent.

**Reply by ThatXliner** [reply]
so, `self.emit`? how do we receive the event?

**Reply by j1o1h1n** [reply]
Have a look at the calculator example and the Button class in _button.py

The Button widget emits an event on click:

```
class Button(Widget):
...
    async def on_click(self, event: events.Click) -> None:
        event.prevent_default().stop()
        await self.emit(ButtonPressed(self))
```

And the app has a handle_button_pressed method:

```
class Calculator(GridView):
    """A working calculator app."""
...
    def handle_button_pressed(self, message: ButtonPressed) -> None:
        assert isinstance(message.sender, Button)
        button_name = message.sender.name

        if button_name.isdigit():
            self.display = self.value = self.value.lstrip("0") + button_name
```
